python/ray/serve/pipeline/api.py

```diff
@@ -63,6 +64,8 @@ def build(ray_dag_root_node: DAGNode) -> List[Deployment]:
     serve_root_dag = ray_dag_root_node.apply_recursive(transform_ray_dag_to_serve_dag)
     deployments = extract_deployments_from_serve_dag(serve_root_dag)
     deployments_with_http = process_ingress_deployment_in_serve_dag(deployments)
+    # Clean up class level deployment name states between build() calls
+    DeploymentNameGenerator.reset()
```

Comment:
this looks like it should be a context manager instead of a global singleton BTW

Reply:
+1

Reply:
hmmm upon second thought: this will be perfect in the InputNode context manager. but we also encouraged users to call .bind() before the input node.

Reply:
^ yes we need something that we know will last for the entire duration of dag binding and transformation